azurerm/internal/services/datamigration/resource_arm_data_migration_service.go

```diff
@@ -0,0 +1,241 @@
+package datamigration
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/Azure/azure-sdk-for-go/services/datamigration/mgmt/2018-04-19/datamigration"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataMigrationService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataMigrationServiceCreate,
+		Read:   resourceArmDataMigrationServiceRead,
+		Update: resourceArmDataMigrationServiceUpdate,
+		Delete: resourceArmDataMigrationServiceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"virtual_subnet_id": {
```

Comment:
Could we rename this to
```suggestion
			"subnet_id": {
```

to match other resources? no where else do we use the term `virtual_subnet_id`